# Illegal Argument Exception test with Botsing

The log come from the following issue: https://github.com/ow2-proactive/programming/issues/1304

## Log treatment before execution

The original log file was separated into three log files illegalArgumentException1.log, illegalArgumentException2.log and illegalArgumentException3.log due to the different `caused by` section. 

illegalArgumentException1.log originally contained the following lines:
```
    at functionalTests.dataspaces.TestDataSpaces$TestActiveObject.readDefaultInputFile(pnp://192.168.122.1:51398/ActiveObject_functionalTests.dataspaces.TestDataSpaces$TestActiveObject_67892988-14ff5d02c06--7ffc--70ca543b814ad80a-67892988-14ff5d02c06--8000)
    at (...)(Unknown Source)
```
Those lines were generated by the java application and are not supported by Botsing. The following message is displayed:

```
17:53:26.303 [main] WARN  eu.stamp.botsing.StackTrace - Missing line in frame 11
17:53:26.304 [main] WARN  eu.stamp.botsing.StackTrace - Missing line in frame 12
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
        at eu.stamp.botsing.StackTrace.stringToStackTraceElement(StackTrace.java:125)
        at eu.stamp.botsing.StackTrace.setup(StackTrace.java:74)
        at eu.stamp.botsing.CrashProperties.setupStackTrace(CrashProperties.java:190)
        at eu.stamp.botsing.Botsing.setupStackTrace(Botsing.java:127)
        at eu.stamp.botsing.Botsing.parseCommandLine(Botsing.java:64)
        at eu.stamp.botsing.Botsing.main(Botsing.java:163)
```

## Execution results

### Execution on illegalArgumentException1.log 

illegalArgumentException1.log was run with target frame from 10 to 1. The execution result is contained in [illegalArgumentException1Result.txt](illegalArgumentException1Result.txt).
With target frame equals to 1, the test [RequestImpl_ESTest.java](generatedTest/org/objectweb/proactive/core/body/request/RequestImpl_ESTest.java) was generated. This test is empty.

### Execution on illegalArgumentException2.log

illegalArgumentException1.log was run with target frame from 10 to 1. The execution result is contained in [illegalArgumentException1Result.txt](illegalArgumentException2Result.txt).
Botsing was unable to generate test with this log.

### Execution on illegalArgumentException3.log

Unable to execute Botsing on the third log. An exception occurs preventing to get results with target frame from 1 to 3.
```
17:19:56.974 [main] INFO  eu.stamp.botsing.StackTrace - Exception type is detected: Caused by
Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1927)
	at eu.stamp.botsing.StackTrace.stringToStackTraceElement(StackTrace.java:102)
	at eu.stamp.botsing.StackTrace.setup(StackTrace.java:74)
	at eu.stamp.botsing.CrashProperties.setupStackTrace(CrashProperties.java:190)
	at eu.stamp.botsing.Botsing.setupStackTrace(Botsing.java:127)
	at eu.stamp.botsing.Botsing.parseCommandLine(Botsing.java:64)
	at eu.stamp.botsing.Botsing.main(Botsing.java:163)
```
